src/adaptors/curve/index.js

```diff
@@ -1,269 +1,277 @@
-    (el) => el.name === entry.pool
-  )?.swap;
-  let apy = entry.apy < 0 ? 0 : entry.apy;
-  apy = chainString === 'ethereum' ? apy : apy * 100;
+      const tvlUsd = pool.usdTotal;
+      if (!tvlUsd) {
+        continue;
+      }
 
-  const newObj = {
-    // ugly, but some swap pools have same address on diff networks
-    pool: `${id}-${chainString}`,
-    chain: utils.formatChain(chainString),
-    project: 'curve',
-    symbol: utils.formatSymbol(token),
-    tvlUsd: entry.tvl,
-    apy: apy,
+      defillamaPooldata.push({
+        pool: address + '-' + blockchainId,
+        chain: utils.formatChain(blockchainId),
+        project: 'curve',
+        symbol: pool.coins.map((coin) => coin.symbol).join('-'),
+        tvlUsd,
+        apyBase,
+        apyReward: aprCrv + aprExtra,
+        apy: apyBase + aprCrv + aprExtra,
+        rewardTokens,
+        underlyingTokens,
+      });
+    }
   };
-  return newObj;
-};
 
-const topLvl = async (chainString) => {
-  if (chainString === 'ethereum') {
-    // pull data
-    poolStats = await getDataEth();
-  } else {
-    // pull data
-    poolStats = await getDataEVM(chainString);
-  }
-  // build pool objects
-  let data = poolStats.map((el) => {
-    if (pools.tokenMapping[chainString][el.pool] === undefined) {
-      return null;
+  // group Promises by blockchain and feed the llama array
+  const responses = [];
+  for (const [blockchainId, poolSubgraphPromise] of Object.entries(
+    blockchainToPoolSubgraphPromise
+  )) {
+    // we already have ethererum for the crv price
+    if (blockchainId !== 'ethereum') {
+      responses.push(
+        Promise.all([
+          blockchainToPoolPromise[blockchainId],
+          poolSubgraphPromise,
+          gaugePromise,
+          extraRewardPromise,
+        ]).then((poolData) => feedLlama(poolData, blockchainId))
+      );
+    } else {
+      responses.push(
+        Promise.all([
+          poolSubgraphPromise,
+          gaugePromise,
+          extraRewardPromise,
+        ]).then((APYandGaugeData) =>
+          feedLlama([ethereumPools, ...APYandGaugeData], blockchainId)
+        )
+      );
     }
-    return buildPool(el, chainString);
-  });
-
-  data = data.filter((el) => el !== null);
-  return data;
-};
+  }
 
-const main = async () => {
-  let data = await Promise.all([
-    topLvl('ethereum'),
-    topLvl('arbitrum'),
-    topLvl('avalanche'),
-    topLvl('fantom'),
-    topLvl('harmony'),
-    topLvl('optimism'),
-    topLvl('polygon'),
-  ]);
+  // wait for all Group promises to resolve
+  try {
+    await Promise.all(responses);
+  } catch (error) {
+    console.error(error);
+  }
 
-  return data.flat();
+  return defillamaPooldata;
 };
 
 module.exports = {
   timetravel: false,
   apy: main,
-  curvePoolStats: getDataEth,
-  tokenMapping: pools.tokenMapping['ethereum'],
+  // legacy needed by convex-finance adaptor
+  curvePoolStats: legacy.curvePoolStats,
+  tokenMapping: legacy.tokenMapping['ethereum'],
```

Comment:
the 'ethereum' key needs to be removed from here. legacy already provides it like that, so here its currently undefined leading convex-finance to fail